### PR TITLE
Ft all parcel by single user

### PR DESCRIPTION
**Pull request by Hurntre**
#### Description
the first feature displays all parcels created by a single user 
the second feature finds a parcel by its id

#### Type of change

- [ ]
- [ ]

#### How Has This Been Tested?

- [ ] Integrated testing

#### PT-ID

[](url)

#### Screenshots

image

image
#### Questions:

### CODE REVIEW

File: server/controllers/parcel.js [hunk 1]
```diff
@@ -69,4 +69,28 @@ export default class ParcelController {
       });
     }
   }
+
+  static async getParcelByID(req, res) {
+    try {
+      const id = req.params.parcelID;
+      const parcel = await ParcelModel.findOne({ _id: id });
+      if (!parcel) {
```
Comment (vic3king) [inline server/controllers/parcel.js:77]: This should be a middleware. Not sure if it's a rule but I like to keep my controllers free of validation checks. In this case if you did move this to a middleware you can then pass your parcel down to the controller so you don't make a new call.

File: server/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable require-jsdoc */
```
Comment (vic3king) [inline server/controllers/user.js:1]: 👀 Please remove

File: server/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import ParcelModel from '../db/models/parcel';
+// import UserModel from '../db/models/user';
+
+/**
+ *  @class UserController
```
Comment (vic3king) [inline server/controllers/user.js:6]: ditto

**Review by vic3king — CHANGES_REQUESTED** [review]
Just a few minor changes and this should be good. Welldone

File: server/controllers/parcel.js (hunk 1 above)
Comment (Hurntre) [reply server/controllers/parcel.js:77]: This isn't a validation check though. it is a simple check to see if the parcel was found before doing some action. in this case sending a reply.

Comment (vic3king) [reply server/controllers/parcel.js:77]: I get that but that's validation in itself. Seems like a small thing to leave right now. But if our application does grow into something complex, having these on our controllers would look untidy.

```
 /**
     * Verify a user after authentication.
     * @param {object} req Restify request object
     * @param {object} res Restify response object
     * @param {function} next Restify routing callback
     * @return {callback}
     */
    verify(req, res, next) {
        this.service.verify(req, res, next)
            .then(data => data)
            .catch((error) => {
                    res.send(new errors.InternalServerError('Internal Server Error'));
            });
    }

This is what a controller on one of my teams look like.

**Pull request merged**